components/NavBar.tsx

```diff
@@ -2,10 +2,14 @@ import React from 'react'
 import { useAuth0 } from '@auth0/auth0-react'
 import { useRouter } from 'next/router'
 
+import { AbstractUser } from '../types/user'
+
 import NavLink from './NavLink'
 
 type Props = {
-  adminUser: boolean
+  // The adminUser object we expect here is missing some information, which is why
+  // the AbstractUser type is sufficient
+  adminUser: AbstractUser | boolean
```

Comment:
Would it maybe be better to just let it be undefined rather than ORing with a boolean? I see where the NavBar is used you do a `|| false`, so it seems like you could just remove the booleans from both sides and just check for undefined, right?

Reply:
If we were architecting this from scratch I'd agree with you but sadly it's not just false we're expecting but sometimes true... :(

In a non-typescript setting the decision to do that makes sense